repository/inventory_repo.go

```diff
@@ -134,3 +180,108 @@ func (i InventoryRepo) GetKeysForEnvironment(ctx context.Context, env string) (m
 	}
 	return scan, nil
 }
+
+func (i InventoryRepo) BuildNotifications(assets domain.Assets) []domain.SSEMessage {
+	var events []domain.SSEMessage
+	events = append(events, getDeleteEvents(assets.Deleted)...)
+	events = append(events, getCreateEvents(assets.Created)...)
+	// TODO: this currently sends patch notification for all flags
+	// regardless weather they have changed or not
+	events = append(events, getPatchEvents(assets.Patched)...)
+	return events
+}
+
+func getDeleteEvents(m map[string]string) []domain.SSEMessage {
+	res := make([]domain.SSEMessage, 0, len(m))
+	if m == nil {
+		return []domain.SSEMessage{}
+	}
+	for k := range m {
+		if strings.Contains(k, featureVariant) {
+			res = append(res, parseFlagEntry(k, deleteVariant))
+		}
+		if strings.Contains(k, segmentVariant) {
+			res = append(res, parseSegmentEntry(k, deleteVariant))
+		}
+	}
+	return res
+}
+
+func getCreateEvents(m map[string]string) []domain.SSEMessage {
+	res := make([]domain.SSEMessage, 0, len(m))
+	if m == nil {
+		return []domain.SSEMessage{}
+	}
+	for k := range m {
+		if strings.Contains(k, featureVariant) {
+			res = append(res, parseFlagEntry(k, createVariant))
+		}
+		if strings.Contains(k, segmentVariant) {
+			res = append(res, parseSegmentEntry(k, createVariant))
+		}
+	}
+	return res
+}
+
+func getPatchEvents(m map[string]string) []domain.SSEMessage {
+	res := make([]domain.SSEMessage, 0, len(m))
+	if m == nil {
+		return []domain.SSEMessage{}
+	}
+	for k := range m {
+		if strings.Contains(k, featureVariant) {
+			res = append(res, parseFlagEntry(k, patchVariant))
+		}
+		if strings.Contains(k, segmentVariant) {
+			res = append(res, parseSegmentEntry(k, patchVariant))
+		}
+	}
+	return res
+}
+
+func parseFlagEntry(flagString, variant string) domain.SSEMessage {
+	env, id, err := parseFlagString(flagString)
+	if err != nil {
+		log.Error(err)
```

Comment:
For consistency should we be using the Proxy logger instead of importing a third party package here for this one log statement?

Reply:
noted